Ratatouille/fftconvolver.cc

```diff
@@ -177,27 +177,23 @@ bool DoubleThreadConvolver::get_buffer(std::string fname, float **buffer, uint32
 void DoubleThreadConvolver::normalize(float* buffer, int asize) {
     // normalize
     if (!norm) return;
+    float peak = 0.0;
     float gain = 0.0;
     // get gain factor from file buffer
     for (int i = 0; i < asize; i++) {
-        gain = std::max(gain, std::abs( buffer[i])) ;
-    }
-    // apply normalize factor
-    for (int i = 0; i < asize; i++) {
-        buffer[i] /= gain;
-    }
-    // loudness compensation
-    gain = 0.0;
-    // get gain square root factor from file buffer
-    for (int i = 0; i < asize; i++) {
+        peak = std::max(peak, std::abs(buffer[i]));
+
         double v = buffer[i] ;
         gain += v*v;
     }
-    // apply gain square root factor when needed
     if (gain != 0.0) {
         gain = 1.0 / gain;
-
-        for (int i = 0; i < asize; i++) {
+    }
```

Comment:
This guard is not only to avoid division by zero, it also avoid that we multiply the buffer contend with zero.

Reply:
Added a guard for peak = 0 as well.